src/common/snippets/include/snippets/utils.hpp

```diff
@@ -32,6 +32,16 @@ std::vector<size_t> get_node_output_layout(const Node* node);
 inline ov::Dimension get_inner_dim(const ov::PartialShape &shape) { return *(shape.rbegin()); }
 inline ov::Dimension get_outer_dim(const ov::PartialShape &shape) { return *(shape.rbegin() + 1); }
 
+// Non-scalar Constants are tokenized as Parameters inside Subgraph body but some operations with constant inputs
+// should have explicit Constants even if they're non-scalar (Reshape, Transpose, Broadcast)
+// This check returns True if Constant op which is input of this op should be inside Subgraph body
+inline auto constant_input_should_be_inside_body(const std::shared_ptr<ov::Node>& node) -> bool {
```

Comment:
This file should contain generic utils which implementation is independant from current Subgraph state. This particular function is actually smt different - it depends on how Snippets do lowering. This is moslty quility of Subgraph op so I would propose to keep it there.